crates/bevy_ecs/src/system/system_registry.rs

```diff
@@ -0,0 +1,277 @@
+use bevy_utils::HashMap;
+
+use crate::system::{BoxedSystem, Command, IntoSystem};
+use crate::world::{Mut, World};
+// Needed for derive(Component) macro
+use crate::{self as bevy_ecs};
+use bevy_ecs_macros::Resource;
+
+/// Stores systems, so they can be reused and run in an ad-hoc fashion.
+///
+/// Systems are keyed by their [`SystemId`](crate::system::SystemId).
+/// Repeated calls with the same function type will reuse cached state, including for change detection.
+///
+/// Any [`Commands`](crate::system::Commands) generated by these systems (but not other systems), will immediately be applied.
+///
+/// This type is stored as a [`Resource`](crate::system::Resource) on each [`World`], initialized by default.
+/// However, it will likely be easier to use the corresponding methods on [`World`],
+/// to avoid having to worry about split mutable borrows yourself.
+///
+/// # Limitations
+///
+///  - Stored systems cannot be chained: they can neither have an [`In`](crate::system::In) nor return any values.
+///  - Stored systems cannot recurse: they cannot run other systems via
```

Comment:
Is this still true?

Reply:
Yes, because `run_system_by_id` needs to have mutable access to the registry, it's briefly removed from the world and no longer available to any other nested `run_system_by_id` calls. This is no longer true for `run_system` since it is stateless and doesn't use the registry at all.

After doing a test I did conclude that exclusive systems *do* actually work.